### PR TITLE
Jifry 201732120165 search function

### PR DESCRIPTION
**Pull request by hema-001**
Group Information:
IBRAHIM MOHAMED IBRAHIM ISMAIL-201732120159              
Yussuf Mohamed Mahdi-201732120173              
Mohamed Jifry Hazzaly Mohamed-201732120165           
Thero Teto Clive-201732120167

### CODE REVIEW

**Review by spm2020spring — COMMENTED** [review]
Each Pull Request should have a single purpose.

I understand this is not always easy. But let's aim at that.


**Hui**

File: Search.jsp [hunk 1]
```diff
@@ -105,13 +105,16 @@ div
 
 footer
 {
-	background:black;
+	background:dodgerblue;
```
Comment (spm2020spring) [inline Search.jsp:108]: Dodgerblue looks better than black as a footer background?

Hui

File: Search.jsp [hunk 2]
```diff
@@ -121,9 +124,9 @@ footer
 <header>
 <nav>
 <a href="Controller?page=home">Home</a>
-<a href="Controller?page=aboutus">About us</a>
-<a href="Controller?page=contact">Contact</a>
-<a href="Controller?page=admin">Admin</a>
+<!--  <a href="Controller?page=aboutus">About us</a> -->
+<a href="Controller?page=contact">Help & Contact</a>
```
Comment (spm2020spring) [inline Search.jsp]: Use "**Contact us for help**".

**Hui**

File: Search.jsp [hunk 3]
```diff
@@ -144,12 +147,13 @@ footer
 <sql:setDataSource user="root" password="Fathihachellam.1" url="jdbc:mysql://localhost:3306/ooad?autoReconnect=true&&useSSL=false" driver="com.mysql.jdbc.Driver" var="db"/>
 
 <sql:query var="result" dataSource="${ db}">
-select * from article where title like '%<%=request.getAttribute("search2") %>%'
+select * from article where title like '%<%=request.getAttribute("search2") %>%' or abstracts like '%<%=request.getAttribute("search2") %>%' or highlight like '%<%=request.getAttribute("search2") %>%' order by time desc
```
Comment (spm2020spring) [inline Search.jsp:150]: What does `search2 `mean?  Could you add a comment for explaining `search2`?

**Hui**

File: Search.jsp [hunk 4]
```diff
@@ -158,6 +162,7 @@ select * from article where title like '%<%=request.getAttribute("search2") %>%'
 <c:forEach items="${result.rows}" var="row">
 <c:set value="${ search2}" var="search2"></c:set>
 <tr>
+<td><c:out value="${ row.subject}"></c:out></td>
```
Comment (spm2020spring) [inline Search.jsp]: Should be `${row.subject}`, no space between `${` and `row`.

**Hui**

File: Search.jsp [hunk 5]
```diff
@@ -167,8 +172,37 @@ select * from article where title like '%<%=request.getAttribute("search2") %>%'
...
+<td><a href="Controller?page=view-article&title=${ row2.title}"> <c:out value="${ row2.title}"></c:out> </a></td>
+<td><c:out value="${ row2.comment}"></c:out></td>
+<td><c:out value="${ row2.time}"></c:out></td>
+<td><c:set var = "string" value = "${ fn:length(row2.user)}"/><c:out value="${ fn:substring(row2.user, 0, 3)}***${ fn:substring(row2.user, string-8, string)}"></c:out></td>
+</tr>
+</c:forEach>
+<tr>
+</tr>
+</table>
+</div>
```
Comment (spm2020spring) [inline Search.jsp]: What is purpose of this block?  Add a comment for that.

**Hui**

File: Search.jsp [hunk 6]
```diff
@@ -167,8 +172,37 @@ select * from article where title like '%<%=request.getAttribute("search2") %>%'
...
+
+
 <footer>
-2019 copyright&copy;mjh.mohamed
+2020 copyright&copy;mjh.mohamed
```
Comment (spm2020spring) [inline Search.jsp]: Copyright (C) 2019, 2020 [Your full name]

Discuss with your team whether or not they want to add their names too.  

They need incentive to do their work.  Credit attribution is a great one.

**Hui**

**Comment by hema-001** [comment]
Hi, @spm2020spring 

Would you please send an invitation to Jifry https://github.com/mjhmohamed , so he can reply to your comments since he is the one made the changes.

-IBRAHIM

**Comment by hema-001** [comment]
> Each Pull Request should have a single purpose.
> 
> I understand this is not always easy. But let's aim at that.
> 
> **Hui**

I did create two separate branches each has its own pull request, one for quiz 4, and this one for the improvement.
You have already merged the first. 
However, i don't understand why all commits are shown in this branch.

-IBRAHIM

**Comment by spm2020spring** [comment]
> Hi, @spm2020spring
> 
> Would you please send an invitation to Jifry https://github.com/mjhmohamed , so he can reply to your comments since he is the one made the changes.
> 
> -IBRAHIM

Done.

Hui

File: Search.jsp (hunk 1 above)
Comment (mjhmohamed) [reply Search.jsp:108]: Yes, it looks better.

File: Search.jsp (hunk 2 above)
Comment (mjhmohamed) [reply Search.jsp]: OK

File: Search.jsp (hunk 3 above)
Comment (mjhmohamed) [reply Search.jsp:150]: It's a string that passes the search keyword through HTTP request from the servlet to the web page.

File: Search.jsp (hunk 4 above)
Comment (mjhmohamed) [reply Search.jsp]: I believe it is not going to affect anything.

File: Search.jsp (hunk 5 above)
Comment (mjhmohamed) [reply Search.jsp]: I'll commit some comments soon.

File: Search.jsp (hunk 6 above)
Comment (mjhmohamed) [reply Search.jsp]: We are thinking to include the team name. But, unfortunately, we don't have one yet.

File: Search.jsp (hunk 3 above)
Comment (spm2020spring) [reply Search.jsp:150]: > It's a string that passes the search keyword through HTTP request from the servlet to the web page.

Thanks @mjhmohamed .  Could you say that in `Search.jsp`, i.e., make comments for `search2 `?

File: Search.jsp (hunk 4 above)
Comment (spm2020spring) [reply Search.jsp]: > I believe it is not going to affect anything.

@mjhmohamed 

Functionally speaking, yes.  But it violates Code Style.   

**Hui**

File: Search.jsp (hunk 3 above)
Comment (mjhmohamed) [reply Search.jsp:150]: will do.

File: Search.jsp (hunk 4 above)
Comment (mjhmohamed) [reply Search.jsp]: OK. I'll try to not violate the Code Style.

File: Search.jsp [hunk 7]
```diff
@@ -202,7 +209,7 @@ select * from comments where comment like '%<%=request.getAttribute("search2") %
 
 
 <footer>
-2020 copyright&copy;mjh.mohamed
+Copyright (C) 2019, 2020 [MIJC Team]
```
Comment (spm2020spring) [inline Search.jsp:212]: @hema-001 

Thanks, MIJC Team。 Do you have a link to the detail of the team (e.g., its members)?

**Hui**

Comment (hema-001) [reply Search.jsp:212]: For now!, NO.

But there will be an improvement to the code structure (i.e., refactoring) soon, I will include a link to the team detail that time.

IBRAHIM

**Pull request merged**